From: David Cohen <david.a.cohen@intel.com>
Subject: extcon: dc-pwrsrc: HACK: SDP connection consumes 500ma from beginning

This patch is a temporary hack while a proper solution comes. The charger
should wait for USB enumeration prior to start to consume 500ma. But for now
we'll go to 500ma directly.

---
 drivers/extcon/extcon-dc-pwrsrc.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- linux.orig/drivers/extcon/extcon-dc-pwrsrc.c
+++ linux/drivers/extcon/extcon-dc-pwrsrc.c
@@ -86,7 +86,12 @@
 
 #define DC_XPWR_CHARGE_CUR_DCP		2000
 #define DC_XPWR_CHARGE_CUR_CDP		1500
-#define DC_XPWR_CHARGE_CUR_SDP		100
+/*
+ * HACK: USB2 SDP connection supposes to start in 100ma and raise up to 500ma
+ * after enumeration. We are still defining a clean standard interface for it.
+ * In meanwhile, we'll go to 500ma right away.
+ */
+#define DC_XPWR_CHARGE_CUR_SDP		500
 
 #define DC_PWRSRC_INTR_NUM		4
 #define PWRSRC_DRV_NAME			"dollar_cove_pwrsrc"
